### PR TITLE
Add telemetry support to atoumtask

### PR DESCRIPTION
**Pull request by oxman**
Add telemetry support to AtoumTask for Phing

### CODE REVIEW

**Comment by Grummfy** [comment]
Thanks for the PR.

We should probably add a check to be sure that atoum report extension is available before ?

Otherwise seems good.

**Comment by oxman** [comment]
Already done : https://github.com/atoum/atoum/pull/612/files#diff-ab8a2832f2790fe82b1aca7e5afb66f7R183

File: resources/phing/AtoumTask.php [hunk 1]
```diff
@@ -171,6 +178,21 @@ public function execute()
 			}
 		}
 
+		if ($this->telemetryEnabled())
+		{
+			if (class_exists('\mageekguy\atoum\reports\telemetry') === false) {
+				throw new exception('AtoumTask depends on atoum/reports-extension being installed to enable telemetry report');
```
Comment (jubianchi) [inline resources/phing/AtoumTask.php]: I would not thrown an exception here to make the dependency optional.

File: resources/phing/AtoumTask.php [hunk 2]
```diff
@@ -171,6 +178,21 @@ public function execute()
...
+			}
+
+			$telemetry = new \mageekguy\atoum\reports\telemetry();
```
Comment (jubianchi) [inline resources/phing/AtoumTask.php]: this whole block should be in the `if`

File: resources/phing/AtoumTask.php (hunk 1 above)
Comment (oxman) [inline resources/phing/AtoumTask.php]: The dependency is not optional if you enabled telemetry, its a mandatory.  
But if you don't enable telemetry the dependency is optional.

File: resources/phing/AtoumTask.php (hunk 2 above)
Comment (oxman) [inline resources/phing/AtoumTask.php]: Why ?

File: resources/phing/AtoumTask.php (hunk 1 above)
Comment (jubianchi) [inline resources/phing/AtoumTask.php]: you are totally right I misread your code

File: resources/phing/AtoumTask.php (hunk 2 above)
Comment (jubianchi) [inline resources/phing/AtoumTask.php]: nevermind

**Comment by Grummfy** [comment]
@jubianchi now it's seems ok, so we can merge?

File: resources/phing/AtoumTask.php (hunk 1 above)
Comment (jubianchi) [inline resources/phing/AtoumTask.php]:
```diff
- AtoumTask depends on atoum/reports-extension being installed to enable telemetry report
+ AtoumTask requires atoum/reports-extension to enable telemetry report
```

File: resources/phing/AtoumTask.php (hunk 2 above)
Comment (jubianchi) [inline resources/phing/AtoumTask.php]: @oxman could you use the short name for this class: `reports\telemetry` ? We already have the required `use` statement.

File: resources/phing/AtoumTask.php [hunk 3]
```diff
@@ -171,6 +178,21 @@ public function execute()
...
+			$telemetry->addWriter(new \mageekguy\atoum\writers\std\out());
```
Comment (jubianchi) [inline resources/phing/AtoumTask.php]: same thing here: `atoum\writers\std\out`

File: resources/phing/AtoumTask.php [hunk 4]
```diff
@@ -287,13 +309,37 @@ public function getCodeCoverage()
 		return $this->codeCoverage;
 	}
 
+	public function setTelemetry($telemetry)
```
Comment (jubianchi) [inline resources/phing/AtoumTask.php]: I would have split this into two method: 
* `enableTelemetry()`
* `disableTelemetry()`

File: resources/phing/AtoumTask.php [hunk 5]
```diff
@@ -287,13 +309,37 @@ public function getCodeCoverage()
...
+	{
+		$this->telemetry = (boolean) $telemetry;
+
+		return $this;
+	}
+
+	public function getTelemetry()
```
Comment (jubianchi) [inline resources/phing/AtoumTask.php]: `TelemetryIsEnabled()` to be consistent with what we do in other class (e.g https://github.com/atoum/atoum/blob/master/classes/test.php#L719)

File: tests/units/resources/phing/AtoumTask.php [hunk 6]
```diff
@@ -543,5 +547,25 @@ public function testSetCodeCoverageCloverPath()
...
 			;
 		}
+
+		public function testGetSetTelemetry()
+		{
+			$this
+				->given($task = new testedClass())
+				->then
+					->object($task->setTelemetry(true))->isIdenticalTo($task)
+					->boolean($task->getTelemetry())->isEqualTo(true)
```
Comment (jubianchi) [inline tests/units/resources/phing/AtoumTask.php]:
```diff
- ->isEqualTo(true)
- ->isTrue
```

File: tests/units/resources/phing/AtoumTask.php [hunk 7]
```diff
@@ -543,5 +547,25 @@ public function testSetCodeCoverageCloverPath()
...
+			;
+		}
+
+		public function testGetSetTelemetryProjectName()
+		{
+			$this
+				->given($task = new testedClass())
+				->then
+				->object($task->setTelemetryProjectName($projectName = uniqid() . '/' . uniqid()))->isIdenticalTo($task)
```
Comment (jubianchi) [inline tests/units/resources/phing/AtoumTask.php]: this line should be indented

File: tests/units/resources/phing/AtoumTask.php [hunk 8]
```diff
@@ -543,5 +547,25 @@ public function testSetCodeCoverageCloverPath()
...
+				->string($task->getTelemetryProjectName())->isEqualTo($projectName)
```
Comment (jubianchi) [inline tests/units/resources/phing/AtoumTask.php]: this line should be indented

File: tests/units/resources/phing/AtoumTask.php [hunk 9]
```diff
@@ -543,5 +547,25 @@ public function testSetCodeCoverageCloverPath()
...
 		}
+
+		public function testGetSetTelemetry()
+		{
+			$this
+				->given($task = new testedClass())
+				->then
+					->object($task->setTelemetry(true))->isIdenticalTo($task)
+					->boolean($task->getTelemetry())->isEqualTo(true)
+			;
```
Comment (jubianchi) [inline tests/units/resources/phing/AtoumTask.php]: you are not testing the case where we disable the telemetry

**Comment by jubianchi** [comment]
@oxman some things left to do:

* I reviewed some bits of the code, could you fix it please?
* Could you also add an entry to the changelog pointing to your PR with credits to you
* Could you please sign your commit (optional but would be appreciated)

File: resources/phing/AtoumTask.php (hunk 4 above)
Comment (oxman) [inline resources/phing/AtoumTask.php]: Not conform to other test

File: resources/phing/AtoumTask.php (hunk 5 above)
Comment (oxman) [inline resources/phing/AtoumTask.php]: Not conform to other test in this file

**Pull request merged**